src/state/slices/opportunitiesSlice/resolvers/foxfarming.ts

```diff
@@ -132,14 +246,73 @@ export const foxFarmingLpUserDataResolver = async ({
   opportunityType: OpportunityDefiType
   accountId: AccountId
   reduxApi: ReduxApi
-}): Promise<{ data: string }> => {
+}): Promise<void> => {
   const { getState } = reduxApi
   const state: ReduxState = getState() as any
   const portfolioLoadingStatusGranular = selectPortfolioLoadingStatusGranular(state)
+
+  // Reject RTK query if account portfolio data is granularily loading
   if (portfolioLoadingStatusGranular?.[accountId] === 'loading')
     throw new Error(`Portfolio data not loaded for ${accountId}`)
 
   const balances: PortfolioAccountBalancesById = selectPortfolioAccountBalances(state)
 
-  return { data: balances[accountId][opportunityId as AssetId] }
+  const hasPortfolioData = Boolean(balances[accountId][opportunityId as AssetId])
+
+  // Reject RTK query if there's no account portfolio data for this LP token
+  if (!hasPortfolioData) {
+    throw new Error('no portfolio data')
+  }
+
+  // All checks passed, resolve the promise so we continue the RTK query execution and populate LP/Account IDs
+  return
+}
+
+export const foxFarmingStakingUserDataResolver = async ({
+  opportunityId,
+  opportunityType: _opportunityType,
+  accountId,
+  reduxApi,
+}: {
+  opportunityId: LpId | StakingId
+  opportunityType: OpportunityDefiType
+  accountId: AccountId
+  reduxApi: ReduxApi
+}): Promise<{ data: GetOpportunityUserStakingDataOutput }> => {
+  const { getState } = reduxApi
+  const state: any = getState() // ReduxState causes circular dependency
+  const assets: AssetsState = state.assets
+  const lpAssetPrecision = assets.byId[foxEthLpAssetId].precision
+  const foxPrecision = assets.byId[foxAssetId].precision
+  const lpTokenMarketData: MarketData = selectMarketDataById(state, foxEthLpAssetId)
+  const lpTokenPrice = lpTokenMarketData?.price
+
+  const { assetReference: contractAddress } = fromAssetId(opportunityId as AssetId)
+  const { account: accountAddress } = fromAccountId(accountId)
+
+  if (bnOrZero(lpTokenPrice).isZero()) {
+    throw new Error(`Market data not ready for ${foxEthLpAssetId}`)
+  }
+
+  const foxFarmingContract = getOrCreateContract(contractAddress, farmingAbi)
```

Comment:
I'm not sure whether or not we will need to be programmatic for now with UniV2 LP/staking contracts and all the yearn, Idle and the like which are known ahead of time, but it would make a lot of sense to be able to implement new opportunities at a low cost

Reply:
ideally it'd be nice to be able to rely on the one code path to fetch abi's from etherscan (which is already a single point of failure for the whole ecosystem) rather than abi's checked into source control, which makes everything flexible and dynamic in the future, but obviously not for this PR